mmv1/third_party/terraform/services/bigquery/resource_bigquery_table.go

```diff
@@ -289,16 +286,21 @@ func resourceBigQueryTableSchemaIsChangeable(old, new interface{}) (bool, error)
 			}
 		}
 		for key := range mapOld {
-			// all old keys should be represented in the new config
+			// dropping top level columns can happen in-place
 			if _, ok := mapNew[key]; !ok {
-				return false, nil
+				if !topLevel {
+					return false, nil
+				}
+				droppedColumns += 1
+				continue
 			}
 			if isChangable, err :=
-				resourceBigQueryTableSchemaIsChangeable(mapOld[key], mapNew[key]); err != nil || !isChangable {
+				resourceBigQueryTableSchemaIsChangeable(mapOld[key], mapNew[key], false); err != nil || !isChangable {
 				return false, err
 			}
 		}
-		return true, nil
+		// only pure column drops allowed
+		return (droppedColumns == 0) || (len(arrayOld) == len(arrayNew)+droppedColumns), nil
```

Comment:
I think "changeableColumns" would be more precise since changing a column's mode would be allowed alongside column dropped, let me know if you think we should change this to only column drops without other in-place changes (besides adding columns)

> and just check old map size against the sum here instead?

hmm not sure if I get it, if the condition is something like
```
(droppedColumns == 0) || (len(arrayOld) == changeableColumns+droppedColumns)
```

then it wouldn't work correctly, for example when changing `{col1, col2, col3}` to `{col1, col4, col5}` the condition would result in `true` (`changeableColumns=1`, `droppedColumns=2`)

I think we can do something like:

```
newColumns := len(arrayNew) - changeableColumns
return (droppedColumns == 0) || (newColumns == 0), nil
```

what do you think?

Reply:
Let's still include other in-place changes. Naming is hard though - the columns would have the same name but maybe different other attributes. "SameNameColumns"?

Your updated logic looks good.